PUBLISHING.md

```diff
@@ -0,0 +1,393 @@
+If you want to use an SSM or KMS key, you'd write it like this, instead:
+
+```
+signing_keys = { kms = { key_id = "abc-def-123" } }
+```
+
+...or...
+
+```
+signing_keys = { ssm = { parameter = "/my/parameter" } }
+```
+
+### Repo location
+
+#### Uploading your repo
+
+Your repo needs to be accessible to your hosts by URL.
+One good place to store repos is S3; this is how Bottlerocket's official repos are stored.
+(If you want, you can put a CloudFront distribution on top of this to make it accessible even more quickly around the world.)
+You can also store your repo behind any HTTP server; the key part is that the repo is accessible from your host.
+This could mean it's publicly accessible, or only accessible inside a VPC, or something similar.
+
+Let's assume you're using an S3 bucket.
+You just need to sync the built repo, like this.
+(If you're using a repo other than `default`, make sure you change the repo name.)
+
+```shell
+aws s3 sync build/repos/default/latest/ s3://my-bucket/
+```
+
+This syncs the metadata and targets directories of the repo into the root of your bucket.
+You can also sync to a subdirectory of your bucket if desired, for example if you use the bucket for other purposes.
+Just make sure you include that subdirectory in the URL in the next step.
+
+> Note: for production repos, it's safer to sync the targets directory before the metadata directory so that clients aren't pointed to targets they can't download yet.
+
+#### Configuring your repo location
+
+After your repo is uploaded, you can add the location into the repo configuration in your `Infra.toml`.
+This will allow you to use `cargo make repo` to update your existing repo in the future, rather than creating a new one from scratch every time.
+This is important so that your hosts can see all available updates in the repo, not just the latest one.
+
+Inside the repo section of your `Infra.toml` (for example, underneath `[repo.default]`) you'd add something like this:
+
+```toml
+metadata_base_url = "https://example.com/"
+targets_url = "https://example.com/targets/"
+```
+
+(You can use a `file://` URL if you want to update a repo based on one you keep locally.)
+
+The variant and architecture are automatically added onto the metadata URL, matching the format of the directories inside `build/repos/default/latest`.
+(The targets directories is shared for all variants and architectures, since target files are prefixed with a checksum.)
+
+### Using your repo from a Bottlerocket host
+
+By default, Bottlerocket hosts talk to the project's official repos.
+There are two ways to point your hosts at your own repo - at build time or at run time.
+
+If you're maintaining your own fork of Bottlerocket, you'd probably want to change the settings at build time, so you don't have to change settings for every host you launch.
+If you're just running a few hosts, or don't want to maintain a fork, then it's easier to change settings at run time.
```

Comment:
If you're running your own repo, won't you need to make sure that the hosts have your keys in the root.json on the filesystem instead of the project's official keys?

Reply:
The only way you get the official Bottlerocket root.json in your image now is if you specifically list its URL in your Infra.toml file - by default, it'll generate a role for you and include that, or use whatever you've pointed to with a URL.  This section is trying to describe how the settings in the model will still be pointing at the Bottlerocket repos, which will have used a different root role, and the options you have for changing that.  Does that make sense, am I misunderstanding?

Reply:
That makes a lot of sense, thanks for the explanation!

In the future, should we change the default settings so that it doesn't include the project's official repository unless you've indicated that in your Infra.toml?

Reply:
Yeah, that makes sense.  We'll need fancier ways of plumbing things together to make it work, and it may be easiest after we have out-of-tree variants going.